src/test/java/spoon/test/template/SubstitutionTest.java

```diff
@@ -176,8 +176,13 @@ public void testCreateTypeFromTemplate() {
         assertNotNull(genEnum);
         assertSame(genEnum, factory.Type().get("generated.GenEnum"));
         assertEquals(2, genEnum.getEnumValues().size());
-        assertEquals("GOOD", genEnum.getEnumValues().get(0).getSimpleName());
-        assertEquals("BETTER", genEnum.getEnumValues().get(1).getSimpleName());
+        if("GOOD".equals(genEnum.getEnumValues().get(0).getSimpleName())){
```

Comment:
Yes, I digged deeper into this and found that the flakiness is occurs because of line 234 in JavaRefectionVisitorImpl. It uses getDeclaredFields() which changes the order. (https://docs.oracle.com/javase/7/docs/api/java/lang/Class.html#getDeclaredFields()). While adding the values in enumValues, the order is thus being changed (line 239 in JavaRefectionVisitorImpl).

Reply:
It might make sense to collect enum constants separately, using the `Class#getEnumConstants()` method. This should preverse order.

Would you like to open a PR for that? If not, I'll open an issue for it so we don't forget about it.